kong/plugins/oauth2/access.lua

```diff
@@ -538,7 +538,7 @@ local function issue_token(conf)
 
     -- Check client_id and redirect_uri
     local allowed_redirect_uris, client = get_redirect_uris(client_id)
-    if not (grant_type == GRANT_CLIENT_CREDENTIALS) then
+    if grant_type ~= GRANT_CLIENT_CREDENTIALS then
```

Comment:
while this is nice, it seems a bit unrelated, but I don't think this is a blocker :-)